Update instructions on how to set up a basic cluster with DS V2 (#1754)
neo4j/docs-operations: modules/ROOT/pages/clustering/setup/deploy.adoc

//Remember to update the settings and link below.
See also xref:clustering/settings.adoc[Settings reference] for more detailed descriptions and examples.

> Neo4j 5.22 introduces the discovery service v2 to discover servers in a cluster and to join them.

Author: ```suggestion
Starting with Neo4j 5.23, you get the option to choose between two discovery services: v1 or v2.
```

> You have the option to choose between two discovery services: v1 or v2.

Author: ```suggestion
```

| The address or network interface this machine uses to listen for incoming messages.
Setting this value to `0.0.0.0` makes Neo4j bind to all available network interfaces.
| xref:configuration/configuration-settings.adoc#config_dbms.cluster.discovery.endpoints[`dbms.cluster.discovery.endpoints`] label:deprecated[Deprecated in 5.23]
| The discovery network address for all the members of the cluster, including this server.
The setting must be set to the same value on all cluster members.
The behavior of this setting can be modified by configuring the setting `dbms.cluster.discovery.resolver_type`.
This is described in detail in xref:clustering/setup/discovery.adoc[].
| xref:configuration/configuration-settings.adoc#config_dbms.cluster.discovery.v2.endpoints[`dbms.cluster.discovery.v2.endpoints`] label:new[Introduced in 5.22]
| A comma-separated list of endpoints that a server should contact in order to discover other cluster members. Typically, all cluster members, including the current server, must be specified in this list.
The setting configures the endpoints for **discovery service v2**.

> | xref:configuration/configuration-settings.adoc#config_dbms.cluster.discovery.version[`dbms.cluster.discovery.version`] label:new[Introduced in 5.22]

Author: Release notes say that those configs were introduced in 5.22

> | xref:configuration/configuration-settings.adoc#config_dbms.cluster.discovery.version[`dbms.cluster.discovery.version`] label:new[Introduced in 5.22]

Reviewer: ah right, they were released then, I'm just wondering if it's worth keeping them consistent with the rest of the Lighthouse release version

| "73e9a990-0a97-4a09-91e9-622bf0b239a4" | "73e9a990-0a97-4a09-91e9-622bf0b239a4" | "server03:7687" | "Enabled" | "Available" | ["system", "neo4j"] | ["system", "neo4j"] | []   | []               | []              | "NONE"         | "5.0.0"     |
+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
----

[TIP]
.Startup time
====
The instance may appear unavailable while it is joining the cluster.
If you want to follow along with the startup, you can see the messages in xref:configuration/file-locations.adoc[_neo4j.log_].
====

======

> [role=include-with-Discovery-service-v2 label--new-5.22]

Author: ```suggestion
[role=include-with-Discovery-service-v2 label--new-5.23]
```

======

In this example, three servers named `server01.example.com`, `server02.example.com` and `server03.example.com` are configured.
Neo4j Enterprise Edition is installed on all three servers.
They are configured by preparing xref:configuration/file-locations.adoc[_neo4j.conf_] on each server.
Note that they are all identical, except for the configuration of `server.default_advertised_address`.
To use discovery service v2, you have to set the configuration of `dbms.cluster.discovery.version` to `V2_ONLY`.
Instead of `dbms.cluster.discovery.endpoints`, use `dbms.cluster.discovery.v2.endpoints`.

._neo4j.conf_ on server01.example.com:
[source, properties]
----
server.default_listen_address=0.0.0.0
server.default_advertised_address=server01.example.com

> dbms.cluster.discovery.v2.endpoints=server01.example.com:5000,server02.example.com:5000,server03.example.com:5000

Reviewer 2: The default port for V2 discovery is `6000` instead of `5000`, so I think each of these need changing.
```suggestion
dbms.cluster.discovery.v2.endpoints=server01.example.com:6000,server02.example.com:6000,server03.example.com:6000
```

dbms.cluster.discovery.version=V2_ONLY
initial.dbms.default_primaries_count=3
----

._neo4j.conf_ on server02.example.com:
[source, properties]
----
server.default_listen_address=0.0.0.0
server.default_advertised_address=server02.example.com

> dbms.cluster.discovery.v2.endpoints=server01.example.com:5000,server02.example.com:5000,server03.example.com:5000

Reviewer 2: ```suggestion
dbms.cluster.discovery.v2.endpoints=server01.example.com:6000,server02.example.com:6000,server03.example.com:6000
```

dbms.cluster.discovery.version=V2_ONLY
initial.dbms.default_primaries_count=3
----

._neo4j.conf_ on server03.example.com:
[source, properties]
----
server.default_listen_address=0.0.0.0
server.default_advertised_address=server03.example.com

> dbms.cluster.discovery.v2.endpoints=server01.example.com:5000,server02.example.com:5000,server03.example.com:5000

Reviewer 2: ```suggestion
dbms.cluster.discovery.v2.endpoints=server01.example.com:6000,server02.example.com:6000,server03.example.com:6000
```
